content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t1 register(Because The ConstantBuffer reserved for SRG1 will be bound to b0 register), and     will take over ALL1+. 
+        Texture2D<float4>        m_texSRV1;      // ERROR. There's no tN available to bind this resource to.
+        Texture2D<float4>        m_texSRV2;      // ERROR. There's no tN available to bind this resource to.
+        RWTexture2D<float4>      m_texUAV1;      // ERROR. There's no uN available to bind this resource to.
+    	RWTexture2D<float4>      m_texUAV2;      // ERROR. There's no uN available to bind this resource to.
+        Sampler                  m_sampler1;     // ERROR. There's no sN available to bind this resource to.
+        Sampler                  m_sampler2;     // ERROR. There's no sN available to bind this resource to.
+        ConstantBuffer<MyStruct> m_cb1; // ERROR. There's no bN available to bind this resource to.
+        ConstantBuffer<MyStruct> m_cb2; // ERROR. There's no bN available to bind this resource to.
+    };
+```
+### Example 8: Fix for Example 7.
+The solution to Example 7, is to move the declaration of `m_texSRV_unbounded` to the end of `SRG1`, basically after `m_cb2`.  
+Alternatively, you could create a second `SRG2`, and move the declaration of `m_texSRV_unbounded` to `SRG2` (As the last resource in `SRG2`, in case there are more resources in `SRG2`.)  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        Texture2D<float4>        m_texSRV2;      // Will be bound to t2 register.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u3 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u4 register.
+        Sampler                  m_sampler1;     // Will be bound to s5 register.
+        Sampler                  m_sampler2;     // Will be bound to s6 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b7 register.
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b8 register.
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t9 register, and will take over ALL10+.
+    };
+```
+### Example 9: Similar to Example 5, but with '--unique-idx'
+In example 5, it is shown how to declare one Unbounded Array for each major resource type: SRV(t), UAV(u), Sampler(s) and CBV(b) inside a single SRG.  
+Unfortunately, when using '--unique-idx', this is impossible to do.  
```

Comment:
* Reference the earlier example. Compare it to this example. 
* Join second sentence on the same line as the first sentence, so they are one paragraph. 
* 2nd sentence, switch to active voice.
```suggestion
### Example: Declaring unbounded arrays with `--unique-idx`
The earlier example, [Example: Declaring unbounded arrays](#example-declaring-unbounded-arrays), demonstrates how to declare an unbounded array for each major resource type: SRV(t), UAV(u), Sampler(s) and CBV(b) inside a single SRG. You cannot do this when compiling with the `--use-idx` option. 
```

Reply:
Fixed.